### ISSUE TITLE
[Sitemap] Space added when formatting `Number:Time` with pattern `%1$tH`

### ISSUE THREAD

**Issue by jlaur** [openhab/openhab-core#3722]
Given:

**items/repro.items:**
```javascript
Number:Time Repro_Pattern "Pattern [X%1$tHX]" <time>
```

**sitemaps/sitemaps.items:**
```
sitemap repro label="Repro" {
        Text item=Repro_Pattern
}
```

Set value:
```shell
openhab:update Repro_Pattern "60 min"
```

MainUI:
![image](https://github.com/openhab/openhab-core/assets/19519842/47e6fb0d-7988-4219-add7-629479e54fc5)

But sitemap has a space added:
![image](https://github.com/openhab/openhab-core/assets/19519842/dd17aa38-990f-4072-a5ea-f13dba0b7149)

## Expected Behavior
When using the provided pattern there should not be any spaces added.

## Current Behavior
A space is added at the end. The same is true for the pattern `%1$tH`, for example. However, if both are used, space is only added after the first, so `X%1$tHX%1$tMX` will become "X01 X00X".

## Possible Solution
Possible regression of #3644. See also openhab/openhab-webui#1953.

## Steps to Reproduce
See description.

## Context
This breaks formatting with state pattern `%1$tH:%1$tM`. For this specific pattern a work-around is to use `%1$tR`. This was fixed just in time for RC1 for the Miele@home binding: openhab/openhab-addons#15280, but other bindings (and user configuration) can easily be affected.

I did not observe this problem in 3.4.4, so it seems to be a 4.0 regression.

## Your Environment
* Version used: 4.0M5

**Comment by jlaur** [comment]
FYI @lolodomo, @mherwege

**Comment by jlaur** [comment]
Fixed in RC1, probably same issue as fixed by #3708.
